configmap/parse.go

```diff
@@ -144,6 +146,45 @@ func AsQuantity(key string, target **resource.Quantity) ParseFunc {
 	}
 }
 
+func AsNamespacedName(key string, target *types.NamespacedName) ParseFunc {
+	return func(data map[string]string) error {
+		var (
+			name      string
+			namespace string
+			raw       string
+			ok        bool
+		)
+
+		if raw, ok = data[key]; !ok {
+			return nil
+		}
+
+		v := strings.Split(raw, string(types.Separator))
+		switch len(v) {
+		// No '/' separator.
+		case 1:
+			name = v[0]
```

Comment:
docs for types.NamespacedName say ["NamespacedName comprises a resource name, with a mandatory namespace"](https://github.com/kubernetes/apimachinery/blob/master/pkg/types/namespacedname.go#L23) which makes this feel a bit odd? Do we really want to allow returning invalid NamespacedNames?

Reply:
I agree - fixed

Reply:
I had this fight and lost with Matt before :)
knative accepts things that K8s would deem invalid :)